azavea/cac-tripplanner
# Feature/kak/map tours#1111

## Overview

Implement tour map page and interactions.

 - route from current origin to final destination with other destinations as waypoints
 - show tour destinations in sidebar cards on map page for tour
 - show destination name on waypoint map marker hover
 - highlight waypoint on map on sidebar place card hover
 - permalink to tour directions from tour detail page
 - link to directions to place or place detail page from tour sidebar
 - adds tours to autocomplete and support routing to them on selection
 - support browser back history to return from place directions to tour directions
 - filter tours and multi-destination events by matching destination(s) on 'explore' tab

Also added a convenience script for running the development gulp server outside the VM.

### Demo

Directions:
![image](https://user-images.githubusercontent.com/960264/65456635-bbed0e80-de17-11e9-95d0-46089b4db311.png)

Filtering by isochrone:
![image](https://user-images.githubusercontent.com/960264/65457522-bbee0e00-de19-11e9-958b-7723c8f7a29a.png)


### Notes

The tour sidebar has a fixed height to force the scrollbar to work; this needs a styling fix.

The itinerary steps aren't presented in the design wireframes, although presenting them was noted in #1121; only the place cards show in the sidebar. @lederer should directions details for the full tour show in the app?

The waypoint markers for tours should be different, according to the wireframes, and probably should also have different interactions (click for a popup instead of hover for a smaller popover?). I'll break that out into a separate task.


## Testing Instructions

 * `./scripts/serve-js-dev.sh` outside VM should start gulp development file watcher
 * Clicking directions link to a tour from a list card should go to tour directions (set origin to route)
 * Clicking 'map' link from tour details page should go to tour directions
 * Selecting tour from destination autocomplete should go to tour directions
 * Filtering in explore mode should work as expected for tours and multi-destination events
 * Hovering a place card in tour directions should highlight place on map
 * Hovering tour destination waypoint marker should show place name in popover
 * Clicking directions button on tour place sidebar card should go to directions for that place
 * Clicking browser back button from the place directions (above) should then return to the tour

Closes #1111 
Closes #1121
